### PR TITLE
x-wso2-owner extension addition 

### PR DESCRIPTION
**Pull request by 7slcode7**
### Purpose

To publish analytics data similar to APIM API invokes. Before adding this extension there were some mismatches in the APIM Analytics dashboard values (API version usage, total API usage etc). 

- Displays analytics data correctly in APIM Analytics dashboard tables
- Updates API usage pie chart correctly

Also adds new wso2 carbon certificate under the name apim300.

### Issues

-  Total API invokes for APIM and Microgateway not updated properly in the API usage pie chart (Microgateway API invokes not included in the total count). 

-  Also API invoke count listed as "__unknown__" field value under a separate row. (For imported APIs from APIM)

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

- Ubuntu 18.04.3 LTS
- Ballerina Version (1.1.0)
- Java open JDK version "11.0.5"
- wso2am-analytics-3.1.0-alpha
- wso2am-3.1.0-alpha

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/APIDetailedDTO.java [hunk 1]
```diff
@@ -103,6 +103,8 @@
     private List<String> accessControlRoles = new ArrayList<String>();
     private APIBusinessInformationDTO businessInformation = null;
     private APICorsConfigurationDTO corsConfiguration = null;
+    //provider string (used to set x-wso2-ownerin swagger definition)
```
Comment (praminda) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/APIDetailedDTO.java]:
```suggestion
    // provider string - used to set x-wso2-owner in swagger definition
```

Comment (Rajith90) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/rest/APIDetailedDTO.java]:
```suggestion
    //provider string (used to set x-wso2-owner in swagger definition)
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal [hunk 2]
```diff
@@ -81,26 +81,29 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
     runtime:InvocationContext invocationContext = runtime:getInvocationContext();
     if (isSecured && invocationContext.attributes.hasKey(AUTHENTICATION_CONTEXT)) {
         AuthenticationContext authContext = <AuthenticationContext>invocationContext.attributes[AUTHENTICATION_CONTEXT];
-        requestResponseExecutionDTO.apiCreator = authContext.apiPublisher;
         requestResponseExecutionDTO.metaClientType = authContext.keyType;
         requestResponseExecutionDTO.applicationConsumerKey = authContext.consumerKey;
         requestResponseExecutionDTO.userName = authContext.username;
         requestResponseExecutionDTO.applicationId = authContext.applicationId;
         requestResponseExecutionDTO.applicationName = authContext.applicationName;
-        requestResponseExecutionDTO.userTenantDomain = authContext.subscriberTenantDomain;
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal]: Why we remove the subscriber tenant domain

File: components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal [hunk 3]
```diff
@@ -81,26 +81,29 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
     runtime:InvocationContext invocationContext = runtime:getInvocationContext();
     if (isSecured && invocationContext.attributes.hasKey(AUTHENTICATION_CONTEXT)) {
         AuthenticationContext authContext = <AuthenticationContext>invocationContext.attributes[AUTHENTICATION_CONTEXT];
-        requestResponseExecutionDTO.apiCreator = authContext.apiPublisher;
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal]: Don't we still have to set the api Creator

Comment (praminda) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal]: publisher is already known to MGW. Therefore this value always set using the `@gateway:publisher` value in https://github.com/wso2/product-microgateway/pull/956/files#diff-65f9f6b86ad06860d5207a18519e41f7R105

File: tests/src/test/java/org/wso2/micro/gateway/tests/common/MockAPIPublisher.java [hunk 4]
```diff
@@ -98,7 +98,7 @@ private API populateJson(API api) throws IOException {
         apiJsonObj.put("name", api.getName());
         apiJsonObj.put("version", api.getVersion());
         apiJsonObj.put("context", api.getContext());
-        apiJsonObj.put("provider", api.getProvider());
+        apiJsonObj.put("provider", api.getProvider());
```
Comment (praminda) [inline tests/src/test/java/org/wso2/micro/gateway/tests/common/MockAPIPublisher.java]:
```suggestion
        apiJsonObj.put("provider", api.getProvider());
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal [hunk 5]
```diff
@@ -81,26 +81,32 @@ public function generateRequestResponseExecutionDataEvent(http:Response response
...
         requestResponseExecutionDTO.metaClientType = authContext.keyType;
         requestResponseExecutionDTO.applicationConsumerKey = authContext.consumerKey;
         requestResponseExecutionDTO.userName = authContext.username;
         requestResponseExecutionDTO.applicationId = authContext.applicationId;
         requestResponseExecutionDTO.applicationName = authContext.applicationName;
-        requestResponseExecutionDTO.userTenantDomain = authContext.subscriberTenantDomain;
+        requestResponseExecutionDTO.userTenantDomain = authContext.subscriberTenantDomain;
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/analytics/responsetime_util.bal]:
```suggestion
        requestResponseExecutionDTO.userTenantDomain = authContext.subscriberTenantDomain;
```

**Comment by 7slcode7** [comment]
This pull request closed and created #974  pull request on behalf of this.

**Pull request closed without merging**